### ISSUE TITLE
Remove yiisoft/di dependency

### ISSUE THREAD

**Issue by vjik** [yiisoft/widget#12]
This is not yii-* package, but him have dependecy from `yiisoft/di`.
I think this is not correct and should be fixed.

**Comment by samdark** [comment]
@vjik would you please make a research on how this could be odne?

**Comment by vjik** [comment]
I didn't invented with anything pretty :(

**Variant one**

Remove dependency only from composer.json.
Provider will be used only applications with `yiisoft/di`.

**Variant two**

Add optional bootstrap to container and create new configs:

`bootstrap.php`:

```php
return [
	'yiisoft/widget' => function(Container $container) {
		WidgetFactory::initialize($container);
	},
];
```

In application `index.php`:

```php
$container = new Container(
    require Builder::path('web'),
    require Builder::path('providers-web'),
    require Builder::path('bootstrap-web'),
);
```

Bootstrap should be executed after container configured.

**Comment by devanych** [comment]
Closed by #31
